Added gp_resgroup_role view into gp_toolkit.

fairyfar opened the pull request:
Added a new view into the resource manager tool `gp_toolkit` to perform the function that are used frequently:

gp_toolkit.gp_resgroup_role: assigned resource group to roles.

- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

ashwinstar commented:
@SmartKeyerror as you are working on resgroup stuff, please can you evaluate if the resgroup related view added by this PR helpful to merge? (I understand resqueue related one is not useful as plans to take resqueue out of GPDB7).

zxuejing commented on src/backend/catalog/gp_toolkit.sql:
```diff
@@ -1695,6 +1695,34 @@ AS
+--
+--------------------------------------------------------------------------------
+
+CREATE VIEW gp_toolkit.gp_resqueue_config
```
There is a similar view that can show all attributes of a resource queue, But, each attribute as a single row:

> postgres=# select * from pg_resqueue_attributes ;
>   rsqname   |      resname      | ressetting | restypid 
> ------------+-------------------+------------+----------
>  pg_default | active_statements | 20              |        1
>  pg_default | max_cost                | -1               |        2
>  pg_default | min_cost                 | 0                |        3
>  pg_default | cost_overcommit   | 0                |        4
>  pg_default | priority                    | medium     |        5
>  pg_default | memory_limit         | -1               |        6
> (6 rows)

others LGTM!

SmartKeyerror reviewed (approved):
Thanks for your contribution, it will be more convinent to find which role belongs to which resgroup, and those views are look good for me.

If possiable, could you please add a test for it? Thanks. And if you're too busy to do this, let me know that and I'm glad to assist you to do that.

Thanks again.

denalex commented:
@SmartKeyerror - can you please merge this PR if it is ready ?

SmartKeyerror commented:
> @SmartKeyerror - can you please merge this PR if it is ready ?

Sure, but I need to add some test cases on the new view, will merge after doing that.

fairyfar commented:
Update the PR:

1. Removed `gp_resqueue_config` view.
2. Added a new test case into `src/test/isolation2/sql/resgroup/resgroup_views.sql`.

> Thanks for your contribution, it will be more convinent to find which role belongs to which resgroup, and those views are look good for me.
> 
> If possiable, could you please add a test for it? Thanks. And if you're too busy to do this, let me know that and I'm glad to assist you to do that.
> 
> Thanks again.

soumyadeep2007 commented:
This is causing a CI failure: https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main/jobs/resource_group_rhel8/builds/684
Can you please take a look @SmartKeyerror and run a fix through a dev pipeline? The PR pipeline does not run resource group jobs.

SmartKeyerror commented:
> This is causing a CI failure: https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main/jobs/resource_group_rhel8/builds/684 Can you please take a look @SmartKeyerror and run a fix through a dev pipeline? The PR pipeline does not run resource group jobs.

Ops, will fix it latter.
